_episodes/31.md

```diff
@@ -0,0 +1,583 @@
+communication between different services. This service is the default when you
+don't specify a `type` value under `spec` in the configuration.
+2. NodePort - is used to expose the internal address of a pod using the IP 
+address and port of the node it is running on.
+3. Load Balancer - this service creates a load balancer for the application in 
+supported cloud providers. We won't cover this one, but this is used when 
+we create our cluster in EKS using the `eksctl`.
+
+Here's a diagram of the ClusterIP networking between different ReplicaSets.
+<p align="center">
+ <img align="center" width="75%" height="100%" src="/assets/episode/31/clusterip.png"/><br/>
+ Source: https://www.udemy.com/course/learn-kubernetes/
+</p>
+
+NodePort's establish connectivity to a specific ReplicaSet of pod instances. It 
+cannot make a generically accessible IP address for services to communicate 
+between one another.
+
+In our case, we configure an external IP address for the coordinator.
+The Helm chart defines a ClusterIP service to accomplish this. Notice the
+selector targets the Trino app, the release label, and only the coordinator 
+component, which we know is one node.
+
+```
+apiVersion: v1
+kind: Service
+metadata:
+  name: tcb-trino
+  labels:
+    app: trino
+    chart: trino-0.3.0
+    release: tcb
+    heritage: Helm
+spec:
+  type: ClusterIP
+  ports:
+    - port: 8080
+      targetPort: http
+      protocol: TCP
+      name: http
+  selector:
+    app: trino
+    release: tcb
+    component: coordinator
+```
+
+### NodePort
+
+The [_NodePort_](https://kubernetes.io/docs/concepts/services-networking/service/#type-nodeport) 
+Service type, creates a proxy service to forward traffic to a specific port on 
+the node from the pod.
+
+<p align="center">
+ <img align="center" width="75%" height="100%" src="/assets/episode/31/service.png"/><br/>
+ Source: https://www.udemy.com/course/learn-kubernetes/
+</p>
+
+There are three ports when setting up a NodePort. 
+
+1. TargetPort - is the port number on the pod itself, where the service forwards to.
+2. Port - is the port used by the service.
+3. NodePort - is the port that is exposed by the worker node and made available 
+externally. NodePorts can only be in the range of 30000 - 32767.
+
+The only required port to set is `port`. By default `targetPort` is the 
+same as `port` and nodePort is automatically assigned a free port in the 
+allowed range. `ports` is also an array which is why the `-` char is used. 
+
+
+### Amazon EKS (Elastic Kubernetes Service)
+
+Amazon EKS is a managed container service to run and scale Kubernetes 
+applications in the cloud or on-premises. EKS manages the control plane for you 
+so you only have to worry about deployment and nodes. Unlike with your own k8s
+cluster, you can't log into the control plane node in EKS, although you won't
+need to. You are able to access workers which are usually EC2 nodes.
```

Comment:
The workers are just pods in the cluster .. what are you trying to say here?

Reply:
I'm actually a little confused by their question. Workers run pods, they are not pods. I'm trying to convey that workers are simply nodes, and in AWS, those nodes are EC2 nodes.